### PR TITLE
Update Typescript Definitions

### PR DESCRIPTION
**Pull request by shakiba**
I'm trying to see if it makes sense to update d.ts files and use classes instead of interface+let. This makes api docs and other things simpler. Downside is that classes needs to be initialized with new, for example `new Ver2()` instead of `Vec2()`.

Could we still find a way to support `planck.Vec2()` (but not `Vec2()` anymore)?

Here is how [API Doc](https://github.com/shakiba/planck.js/blob/790a15775d6e4af4ac7f88fc9d33f8b79bb6a461/docs/README.md) looks like after.

### CODE REVIEW

**Comment by zOadT** [comment]
Indeed there is [a solution](https://devblogs.microsoft.com/typescript/announcing-typescript-3-6/#ambient-classes-and-functions-can-merge) since Typescript 3.6. This was on my TODO list for some time now, but I forgot 😅

**Comment by shakiba** [comment]
Noice! I will do it soon!

File: lib/common/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export let Math: Math & {
+export class Math /*extends JSMath*/ {
```
Comment (shakiba) [inline lib/common/index.d.ts:1]: @zOadT do you also have any idea about this?

Comment (shakiba) [reply lib/common/index.d.ts:1]: if I use `class Math extend Math` it will error.

Comment (zOadT) [reply lib/common/index.d.ts:1]: Hmm currently not. Does the old method generate weird docs?

I can only think of manually setting the required JSMath methods for now 😕

Comment (zOadT) [reply lib/common/index.d.ts:1]: Oh, maybe you can just name is PlanckMath and do export {PlanckMath as Math} but I can't test it right now

Comment (shakiba) [reply lib/common/index.d.ts:1]: yeah, that looks great, I was also thinking about something like that

Comment (zOadT) [reply lib/common/index.d.ts:1]: Wait a sec, does using class actually make sense here? We won't create an instance and by inheritance we don't get the desired static methods on PlankMath, do we?

Comment (shakiba) [reply lib/common/index.d.ts:1]: good point, I made functions static.
I wasn't able to do `export {PlanckMath as Math}`, please feel free to try it

**Pull request merged**